client/src/main/kotlin/io/provenance/objectstore/gateway/client/GatewayClient.kt

```diff
@@ -31,28 +41,42 @@ class GatewayClient(val config: ClientConfig): Closeable {
 
     private val gatewayStub = GatewayGrpc.newFutureStub(channel)
 
+    fun createJWT(keyPair: KeyPair, expiresAt: OffsetDateTime = OffsetDateTime.now().plusSeconds(60)): String = JWT.create()
+        .withIssuedAt(OffsetDateTime.now().toInstant().let(Date::from))
+        .withExpiresAt(expiresAt.toInstant().let(Date::from))
+        .withIssuer("object-store-gateway")
+        .withClaim("sub", keyPair.public.let(ECUtils::publicKeyEncoded))
+        .withClaim("addr", keyPair.public.getAddress(config.mainNet))
+        .sign(Algorithm.ECDSA256K(keyPair.public as ECPublicKey, keyPair.private as ECPrivateKey))
+
+    /**
+     * Fetch scope data from gateway, creating a fresh JWT for authentication
+     * @param scopeAddress the scope's address
+     * @param keyRef the KeyRef of the key to sign the request with/decrypt the received response
+     * @param timeout an optional timeout for the request/used in the request to expire signature
+     */
+    fun requestScopeData(scopeAddress: String, keyPair: KeyPair, timeout: Duration = Duration.ofSeconds(10)): GatewayOuterClass.FetchObjectResponse {
```

Comment:
This is nice and easy to use! Excellent

Reply:
hopefully you will only have to change from a `KeyRef` to a `KeyPair` from the verifier... unfortunately there didn't seem to be any way that I could use our signer to sign the JWT, so I needed access to the private key

Reply:
nevermind on that, figured out a way to add in my own Algorithm class to use for generating the JWT, so we can deal with a KeyRef still, so the function signature can stay the same